### ISSUE TITLE
scsynth: adding yaml config parsing to cli interface

### ISSUE THREAD

**Issue by LFSaw** [supercollider/supercollider#3118]
stub issue, promised to be filled by @brianlheim :)

**Comment by brianlheim** [comment]
this is really a couple of things at once:

- add an option to specify a yaml file to be parsed for config
- add an option to specify paths to exclude from loading (the inverse of `-U` (I think that's the option))
- refactor sc_languageconfig stuff in the language component so that it can be used for both server config and language config parsing
- modify scsynth's loading routine so that it loads from these paths

kinda depends on #2861

**Comment by vivid-synth** [comment]
I'm not a huuge fan of YAML (giant spec, almost all implementations other than the C reference have bugs and/or undefined behavior)... Currently the servers don't have a dependency on it, and it might be nice to keep it that way.

Other options:
  - JSON (looks like we already depend on ```simplejson```)
  - If we really are just storing key-value pairs, perhaps just:

```
--udp-port 57110
--samplerate 96000
--max-logins 64
```

?

**Comment by miguel-negrao** [comment]
> refactor sc_languageconfig stuff in the language component so that it can be used for both server config and language config parsing

And also de IDE ! It would be good to have the lang yaml parsing code implemented just once, so as not to suffer from bit rot when it is updated in just one place and not updated in another, as happened recently with the ide code.

**Comment by brianlheim** [comment]
> I'm not a huuge fan of YAML (giant spec, almost all implementations other than the C reference have bugs and/or undefined behavior)... Currently the servers don't have a dependency on it, and it might be nice to keep it that way.

sclang config is already in yaml, so this is the least breaking change that allows both to be configured with the same document. i should have clarified that "to be parsed for config" means only path-related config, i.e. just reading `sclang_conf.yml`.

also, the simplejson lib here is python, not C++.

if you would like to take the time to bring in a json dependency and add the parsing code to server startup, go ahead, i don't think it's a bad idea, but i would prefer to do it in yaml.

**Comment by brianlheim** [comment]
I've updated the issue title to more accurately reflect what we're after. We can also definitely talk about adding a general CLI config file capability in a separate issue, but this one is just about specifying include/exclude paths to match sclang's compilation behavior.

**Comment by brianlheim** [comment]
also @vivid-synth - b::po gives config file parsing almost for free (!), with options in `key=value`, .cfg-style format: http://www.boost.org/doc/libs/1_64_0/doc/html/program_options/overview.html#idp435878016

**Comment by miguel-negrao** [comment]
@brianlheim in order to centralize the yaml parsing code (at the moment between the ide and sclang, later scsynth too), I thought that it would be better to first get the language config data through the SC_LanguageConfig class and then copy it from there to the IDE objects. I think the extra copying can be justified by the code being more maintainable. What do you think ?

**Comment by vivid-synth** [comment]
The simple ```key=value``` parsing seems good to me!

**Comment by brianlheim** [comment]
@miguel-negrao sorry, just realized i never replied to you. I am not sure I understand what you mean... "copying" in this context means what? and what are the "IDE objects"?
